### DISCUSSION TITLE
GCNConv vs ChebConv

### DISCUSSION THREAD

**Discussion by chinmay5** [pyg-team/pytorch_geometric#2424] [Q&A] [answered]
I am starting with the framework and would appreciate it if someone could help me with certain basic doubts:- 

I have a node classification task and have used a ChebConv and GCNConv operators. While the ChebConv gives me decent results, (acc ~ 0.71), with the same code base and replacement by GCNConv, I am not able to converge the network. Here are some of my doubts and I would really appreciate it if someone could explain them:-

1. ChebConv would get neighborhood information based on the degree **K**. I should get the same effects by using **K** GCNConv layers cascaded. However, this does not seem to happen.
2. Since even the training loss is not decreasing much and I am using the same code base, there should be some specific property of ChebConv which sets it apart from GCNConv. Can someone please explain what that might possibly be?

**Comment by wsad1** [comment, marked as answer]
Its not easy to give a definitive answer here. The difference in accuracy might be down to your data or the number of layers you are using. Here are some thoughts on your 2 questions.

- ChebConv with degree K and GCNConv with K layers aren't the same. Each GCNConv layer is aggregating node embeddings from the previous GCNConv layer. But ChebConv always aggregates raw node embeddings.

- GCNConv tends to over-smooth representations when there are many layers. But ChebConv can potentially avoid that because of how its output is a "combination of k-hop node embeddings".

![image](https://user-images.githubusercontent.com/13963626/115355696-8a494e80-a1d8-11eb-973a-596b63ce5f9f.png)

Two things you could try/check.

1. Look at the [homophily](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.homophily) of your labels. If the homophily is low, then that could be the reason why ChebConv works better.
2. Try adding [JKconnections](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.models.JumpingKnowledge) to your GCN that might help with the over-smoothing, if that is indeed what's causing the difference in accuracy.

**Reply by chinmay5** [reply]
Thank you, that helped me a lot. I found out that the homophily of the labels was ~0.5 and hence GCN struggled. Special thanks for the clarification with the first pointer. I replaced the GCN with SageConv and am currently experimenting with other suggestions but this undoubtedly gives me a better understanding of why ChebConv worked better. Thank you.
